gstreamer-pbutils/src/auto/discoverer.rs

```diff
@@ -0,0 +1,184 @@
+}
+
+impl Discoverer {
+    pub fn new(timeout: gst::ClockTime) -> Result<Discoverer, Error> {
+        assert_initialized_main_thread!();
+        unsafe {
+            let mut error = ptr::null_mut();
+            let ret = ffi::gst_discoverer_new(timeout.to_glib(), &mut error);
+            if error.is_null() { Ok(from_glib_full(ret)) } else { Err(from_glib_full(error)) }
+        }
+    }
+}
+
+unsafe impl Send for Discoverer {}
+unsafe impl Sync for Discoverer {}
+
+pub trait DiscovererExt {
+    fn discover_uri(&self, uri: &str) -> Result<DiscovererInfo, Error>;
+
+    fn discover_uri_async(&self, uri: &str) -> bool;
+
+    fn start(&self);
+
+    fn stop(&self);
+
+    fn get_property_timeout(&self) -> u64;
+
+    fn set_property_timeout(&self, timeout: u64);
+
+    fn connect_discovered<F: Fn(&Self, &DiscovererInfo, &Error) + Send + Sync + 'static>(&self, f: F) -> SignalHandlerId;
+
+    fn connect_finished<F: Fn(&Self) + Send + Sync + 'static>(&self, f: F) -> SignalHandlerId;
+
+    fn connect_source_setup<F: Fn(&Self, &gst::Element) + Send + Sync + 'static>(&self, f: F) -> SignalHandlerId;
+
+    fn connect_starting<F: Fn(&Self) + Send + Sync + 'static>(&self, f: F) -> SignalHandlerId;
+
+    fn connect_property_timeout_notify<F: Fn(&Self) + Send + Sync + 'static>(&self, f: F) -> SignalHandlerId;
+}
+
+impl<O: IsA<Discoverer> + IsA<glib::object::Object>> DiscovererExt for O {
+    fn discover_uri(&self, uri: &str) -> Result<DiscovererInfo, Error> {
+        unsafe {
+            let mut error = ptr::null_mut();
+            let ret = ffi::gst_discoverer_discover_uri(self.to_glib_none().0, uri.to_glib_none().0, &mut error);
+            if error.is_null() { Ok(from_glib_full(ret)) } else { Err(from_glib_full(error)) }
+        }
+    }
+
+    fn discover_uri_async(&self, uri: &str) -> bool {
+        unsafe {
+            from_glib(ffi::gst_discoverer_discover_uri_async(self.to_glib_none().0, uri.to_glib_none().0))
+        }
+    }
+
+    fn start(&self) {
+        unsafe {
+            ffi::gst_discoverer_start(self.to_glib_none().0);
+        }
+    }
+
+    fn stop(&self) {
+        unsafe {
+            ffi::gst_discoverer_stop(self.to_glib_none().0);
+        }
+    }
+
+    fn get_property_timeout(&self) -> u64 {
+        let mut value = Value::from(&0u64);
+        unsafe {
+            gobject_ffi::g_object_get_property(self.to_glib_none().0, "timeout".to_glib_none().0, value.to_glib_none_mut().0);
+        }
+        value.get().unwrap()
+    }
+
+    fn set_property_timeout(&self, timeout: u64) {
+        unsafe {
+            gobject_ffi::g_object_set_property(self.to_glib_none().0, "timeout".to_glib_none().0, Value::from(&timeout).to_glib_none().0);
+        }
+    }
+
+    fn connect_discovered<F: Fn(&Self, &DiscovererInfo, &Error) + Send + Sync + 'static>(&self, f: F) -> SignalHandlerId {
+        unsafe {
+            let f: Box_<Box_<Fn(&Self, &DiscovererInfo, &Error) + Send + Sync + 'static>> = Box_::new(Box_::new(f));
```

Comment:
This is going to explode if DiscovererInfo or the Error are NULL, which AFAIU can both happen.